### PR TITLE
Remove submodule and bring project dependencies up to date

### PR DESCRIPTION
**Pull request by bnsblue**

### CODE REVIEW

File: .github/workflows/pythonpublish.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Upload Python Package
```
Comment (kumare3) [inline .github/workflows/pythonpublish.yaml:1]: does this work?

File: .secrets.baseline [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      "name": "StripeDetector"
+    },
+    {
+      "name": "TwilioKeyDetector"
+    }
+  ],
+  "results": {
+    "config.yaml": [
+      {
+        "hashed_secret": "2f6f0c24ab741f0e2a0ffdd200adca35f2f72f87",
```
Comment (kumare3) [inline .secrets.baseline:63]: what are these?

File: demo/flyte.config [hunk 3]
```diff
@@ -3,12 +3,17 @@ workflow_packages=workflows
...
+assumable_iam_role=arn:aws:iam::173840052742:role/modelbuilderapibatchworker-staging
 
 [platform]
-url=localhost:30081
-insecure=True
+url=flyte-staging.lyft.net
 
 [aws]
-s3_shard_formatter=s3://flytedemo-us-east-2/{}/
+s3_shard_formatter=s3://lyft-modelbuilder/{}/
```
Comment (kumare3) [inline demo/flyte.config:12]: we should not be having lyft internal details here please!!!

File: demo/flyte.config [hunk 4]
```diff
@@ -3,12 +3,17 @@ workflow_packages=workflows
 python_venv=flytekit_venv
 
 [auth]
-assumable_iam_role=cluster-role
+assumable_iam_role=arn:aws:iam::173840052742:role/modelbuilderapibatchworker-staging
```
Comment (kumare3) [inline demo/flyte.config:6]: remove please

File: demo/workflows/sagemaker_xgboost_hpo.py [hunk 5]
```diff
@@ -39,15 +39,16 @@
 }
 
 xgtrainer_task = SagemakerXgBoostOptimizer(
-    region="us-east-2",
-    role_arn="arn:aws:iam::236416911133:role/SageMaker-Executor",
+    region="us-east-1",
+    role_arn="arn:aws:iam::173840052742:role/modelbuilderapibatchworker-staging",
```
Comment (kumare3) [inline demo/workflows/sagemaker_xgboost_hpo.py:43]: remove

File: .github/workflows/pythonpublish.yaml (hunk 1 above)
Comment (bnsblue) [reply .github/workflows/pythonpublish.yaml:1]: nope

File: .secrets.baseline (hunk 2 above)
Comment (bnsblue) [reply .secrets.baseline:63]: I set up https://github.com/Yelp/detect-secrets in this repo. This is the baseline detection result

Comment (bnsblue) [reply .secrets.baseline:63]: These "hashed_secret" listed here are not really secrets. They are either hashed high entropy strings (SHAs) or non-sensitive secrets like `FLYTE_AWS_SECRET_ACCESS_KEY: miniostorage`

File: .github/workflows/pythonpublish.yaml (hunk 1 above)
Comment (bnsblue) [reply .github/workflows/pythonpublish.yaml]: removed

**Pull request merged**